pipelines/controller/src/main/java/org/openmrs/analytics/DWHFilesManager.java

```diff
@@ -0,0 +1,353 @@
+    ResourceId startTimestampResource =
+        dwhResource.resolve(DwhFiles.TIMESTAMP_FILE_START, StandardResolveOptions.RESOLVE_FILE);
+    ResourceId endTimestampResource =
+        dwhResource.resolve(DwhFiles.TIMESTAMP_FILE_END, StandardResolveOptions.RESOLVE_FILE);
+    if (doesFileExist(startTimestampResource) && doesFileExist(endTimestampResource)) {
+      return true;
+    }
+    return false;
+  }
+
+  private boolean doesFileExist(ResourceId resourceId) throws IOException {
+    List<MatchResult> matchResultList =
+        FileSystems.matchResources(Collections.singletonList(resourceId));
+    MatchResult matchResult = Iterables.getOnlyElement(matchResultList);
+    if (matchResult.status() == Status.OK) {
+      return true;
+    } else if (matchResult.status() == Status.NOT_FOUND) {
+      return false;
+    } else {
+      String errorMessage =
+          String.format("Error matching file spec %s: status %s", resourceId, matchResult.status());
+      logger.error(errorMessage);
+      throw new IOException(errorMessage);
+    }
+  }
+
+  /**
+   * This method returns the base directory where the DWH snapshots needs to be created. This is
+   * determined by ignoring the prefix part in the given input format <baseDir></prefix> for the
+   * dwhRootPrefix
+   *
+   * @param dwhRootPrefix
+   * @return the base directory name
+   */
+  public String getBaseDir(String dwhRootPrefix) {
+    int index = dwhRootPrefix.lastIndexOf("/");
+    if (index <= 0) {
+      String errorMessage =
+          "dwhRootPrefix should be configured with a non-empty base directory. It should be of the"
+              + " format <baseDir>/<prefix>";
+      logger.error(errorMessage);
+      throw new IllegalArgumentException(errorMessage);
+    }
+    return dwhRootPrefix.substring(0, index);
+  }
+
+  /**
+   * This method returns the prefix name that needs to be applied to the DWH snapshot root folder
+   * name. This is determined by considering the prefix part in the given input format
+   * <baseDir></prefix> for the dwhRootPrefix
+   *
+   * @param dwhRootPrefix
+   * @return the prefix name
+   */
+  public String getPrefix(String dwhRootPrefix) {
+    int index = dwhRootPrefix.lastIndexOf("/");
+    String prefix = dwhRootPrefix.substring(index + 1);
+    if (prefix == null || prefix.isBlank()) {
+      String errorMessage =
+          "dwhRootPrefix should be configured with a non-empty suffix string after the last"
+              + " occurrence of the character '/'";
+      logger.error(errorMessage);
+      throw new IllegalArgumentException(errorMessage);
+    }
+    return prefix;
+  }
+
+  /**
+   * Returns all the child directories under the given base directory which are 1-level deep.
+   *
+   * @param baseDir
+   * @return The list of all child directories under the base directory
+   * @throws IOException
+   */
+  public Set<ResourceId> getAllChildDirectories(String baseDir) throws IOException {
```

Comment:
For the few functions that are moved from `PipelineManager`, I assumed no code is changed; please let me know if that is not the case.

Reply:
Yes, the methods which are moved from `PipelineManager' are not modified.

Reply:
I have made small refactoring to the methods `getBaseDir` and `getPrefix` which are moved from `PipelineManager', to made them little robust for different file systems.

Reply:
Okay, thanks.